### PR TITLE
Fixes facebook/presto#2040

### PR DESCRIPTION
**Pull request by lstyls**
Updated pull request after rebased to latest

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/MaxBy.java [hunk 1]
```diff
@@ -123,6 +123,10 @@ public static void combine(MaxOrMinByState state, MaxOrMinByState otherState)
             state.setKey(otherState.getKey());
             state.setValue(otherState.getValue());
         }
+        else if (otherState.getKey() == null) {
+            otherState.setKey(state.getKey());
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/MaxBy.java]: this is unnecessary. otherState doesn't need to be updated, so you can just return here

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/MinBy.java [hunk 2]
```diff
@@ -123,6 +123,10 @@ public static void combine(MaxOrMinByState state, MaxOrMinByState otherState)
             state.setKey(otherState.getKey());
             state.setValue(otherState.getValue());
         }
+        else if (otherState.getKey() == null) {
+            otherState.setKey(state.getKey());
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/MinBy.java]: same comment

**Comment by cberner** [comment]
Assign back to me when you've address the comments

**Comment by cberner** [comment]
Ping. Did you still want to get this in?

**Comment by lstyls** [comment]
Hey @cberner, I did something that made github automatically close this PR. I have resubmitted with your changes here: https://github.com/facebook/presto/pull/2627

**Pull request closed without merging**